### PR TITLE
adding little helper tool like icepll

### PR DESCRIPTION
**Pull request by multigcs**
written in python,
only for the GW1N-9C
finds the best match

### CODE REVIEW

File: gowin-pll.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python3
```
Comment (yrabbit) [inline gowin-pll.py:1]: ` #!/usr/bin/env python3`

?

**Comment by yrabbit** [comment]
Great stuff! It generates working code for 9C.
How about adding a board selection parameter like `-d GW1NZ-1` ?

**Comment by multigcs** [comment]
i have added more chip types but i can not test this all, i have only the TangNano9K board

**Comment by yrabbit** [comment]
Great job! I will play with different boards in the next day or two.

**Comment by yrabbit** [comment]
Works for Tangnano4k and Tangnano9k

File: gowin-pll.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-i", "--input-freq-mhz", help="PLL Input Frequency", type=float, default=27
+)
+parser.add_argument(
+    "-o", "--output-freq-mhz", help="PLL Output Frequency", type=float, default=108
+)
+parser.add_argument("-d", "--device", help="Device", type=str, default="GW1NR-9 C6/I5")
```
Comment (yrabbit) [inline gowin-pll.py:19]: Most of our tools accept the chip indication parameter as a part number or as a series name. Having two ways is not good in itself, and we would not want to add a third way of specifying it.

Could you extract the information you need from the part number?
GW1N-UV4LQ144C6/I5
GW1N-LV1QN48C6/I5
GW1NZ-LV1QN48C6/I5
GW1NSR-LV4CQN48PC7/I6
GW1NR-LV9QN88PC6/I5

**Pull request merged**